src/components/common/GProfiler/sagas.ts

```diff
@@ -0,0 +1,46 @@
+import { put, takeLatest, call } from 'redux-saga/effects';
+import * as R from 'ramda';
+
+import * as t from './actionTypes';
+import * as c from './constants';
+import { GProfilerOrganism } from './types';
+import * as Action from './actions';
+
+async function fetchJson(url: string) {
+    let resp = null;
+    try {
+        let data = await fetch(url);
+        resp = { data: await data.json() };
+    } catch (e) {
+        resp = { err: e.message };
+    }
+    return resp;
+}
+
+function* fetchAvailableOrganisms() {
+    yield put(Action.fetchingAvailableOrganisms(true));
```

Comment:
This should probably go in the reducer for `FETCH_AVAILABLE_ORGANISMS`. Doesn't need another action.

Reply:
[Here](https://github.com/reduxjs/redux/blob/master/src/types/reducers.ts#L29) they say: `* *Do not put API calls into reducers.*`

Reply:
Indeed. Maybe I'm misunderstanding but this isn't an API call, it's just flipping the fetching flag in the model to `true`?

Reply:
Ok now I see what you mean
Yeah indeed it's better what you're suggesting